picasso/src/main/java/com/squareup/picasso3/AssetRequestHandler.java

```diff
@@ -44,16 +43,20 @@
         && !uri.getPathSegments().isEmpty() && ANDROID_ASSET.equals(uri.getPathSegments().get(0)));
   }
 
-  @Override public Result load(Request request, int networkPolicy) throws IOException {
+  @Override public void load(Request request, int networkPolicy, Callback callback) {
     if (assetManager == null) {
       synchronized (lock) {
         if (assetManager == null) {
           assetManager = context.getAssets();
         }
       }
     }
-    Source source = Okio.source(assetManager.open(getFilePath(request)));
-    return new Result(source, DISK);
+    try {
+      Source source = Okio.source(assetManager.open(getFilePath(request)));
+      callback.onSuccess(new Result(source, DISK));
+    } catch (Exception e) {
+      callback.onError(e);
```

Comment:
callback could be called twice. did you mean to catch only the IOExceptions from line 55?

Reply:
I think we probably want to design the internal callback API that way so it works more like Rx.

Reply:
updated.

Reply:
@JakeWharton can tackle in a follow-up PR.